### PR TITLE
KeyValue转换器切分方式修改

### PR DESCRIPTION
**Pull request by PapaPiya**
将原先的有logfmt包切分，复用为parse模块里的keyvalue切分方式

### CODE REVIEW

File: transforms/mutate/keyvalue_test.go [hunk 1]
```diff
@@ -93,21 +94,21 @@ func TestKV_Transform(t *testing.T) {
 		{
 			line:       []Data{{"raw": `foo="" bar=`}},
 			expectData: []Data{{"raw": `foo="" bar=`}},
-			expectErr:  errors.New("find total 1 erorrs in transform keyvalue, last error info is no value matched in key value transform in raw"),
+			expectErr:  errors.New("find total 1 erorrs in transform keyvalue, last error info is parse transform key valse failed, error msg: success 0 errors 1 last error key value not match, will keep origin data in pandora_stash if disable_record_errdata field is false, send error detail <nil>"),
```
Comment (redHJ) [inline transforms/mutate/keyvalue_test.go]: valse-> value

File: transforms/mutate/keyvalue_test.go [hunk 2]
```diff
@@ -72,7 +73,7 @@ func TestKV_Transform(t *testing.T) {
 		{
 			line:       []Data{{"raw": `123456789012345`}},
 			expectData: []Data{{"raw": `123456789012345`}},
-			expectErr:  errors.New("find total 1 erorrs in transform keyvalue, last error info is no value matched in key value transform in raw"),
+			expectErr:  errors.New("find total 1 erorrs in transform keyvalue, last error info is parse transform key valse failed, error msg: success 0 errors 1 last error no splitter exist, will keep origin data in pandora_stash if disable_record_errdata field is false, send error detail <nil>"),
```
Comment (redHJ) [inline transforms/mutate/keyvalue_test.go]: valse-> value

File: transforms/mutate/keyvalue_test.go [hunk 3]
```diff
@@ -93,21 +94,21 @@ func TestKV_Transform(t *testing.T) {
...
+			expectData: []Data{{"raw": Data{"abc": "abc", "foo": "\"def"}}},
+			expectErr:  nil,
 			splitter:   "=",
 			new:        "raw",
 		},
 		{
 			line:       []Data{{"raw": `"foo=" bar=abc`}},
 			expectData: []Data{{"raw": `"foo=" bar=abc`}},
-			expectErr:  errors.New(`find total 1 erorrs in transform keyvalue, last error info is logfmt syntax error at pos 1 on line 1: unexpected '"'`),
+			expectErr:  errors.New(`find total 1 erorrs in transform keyvalue, last error info is parse transform key valse failed, error msg: success 0 errors 1 last error no value or key was parsed after logfmt, will keep origin data in pandora_stash if disable_record_errdata field is false, send error detail <nil>`),
```
Comment (redHJ) [inline transforms/mutate/keyvalue_test.go]: valse-> value

File: transforms/mutate/keyvalue.go [hunk 4]
```diff
@@ -1,15 +1,15 @@
 package mutate
 
 import (
-	"bytes"
 	"errors"
+	"fmt"
+	"github.com/qiniu/logkit/conf"
```
Comment (redHJ) [inline transforms/mutate/keyvalue.go]: 顺序

File: transforms/mutate/keyvalue.go [hunk 5]
```diff
@@ -254,55 +254,23 @@ func (k *KV) transform(dataPipeline <-chan transforms.TransformInfo, resultChan
...
-			if bValue, err := strconv.ParseBool(value); err == nil {
-				fields[string(decoder.Key())] = bValue
-				continue
-			}
-			fields[string(decoder.Key())] = value
-		}
-		if len(fields) == 0 {
-			continue
-		}
+	c := conf.MapConf{
```
Comment (redHJ) [inline transforms/mutate/keyvalue.go]: 这边这么实现不好，parser/transform不要直接调用，公用部分封装到utils/parse里

**Comment by PapaPiya** [comment]
parse:keyvalue，transform:  key:time, new:new.time, type: keyvalue
time=date=2020 01 15 18:42 major=math => 
[{"time":"date=2020 01 15 18:42","major":"math","datasource":"/Users/shijinchen/test/logkit/up.log","new":{"time":{"date":"2020 01 15 18:42"}}}]

**Comment by redHJ** [comment]
lgtm

**Pull request merged**